### PR TITLE
Added handling of cancel for adv auth flow - Show Host picker

### PR DESCRIPTION
**Pull request by trooper2013**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m [hunk 1]
```diff
@@ -385,13 +385,23 @@ - (void)oauthCoordinator:(SFOAuthCoordinator *)coordinator willBeginBrowserAuthe
         };
         builder.actionTwoCompletion = ^{
             __strong typeof(weakSelf) strongSelf = weakSelf;
+            BOOL handledByDelegate = NO;
             // Let the OAuth coordinator know whether to proceed or not.
-            if ([strongSelf.config.safariViewDelegate respondsToSelector:@selector(authClientDidCancelBrowserFlow:)]) {
-                [strongSelf.config.safariViewDelegate authClientDidCancelBrowserFlow:strongSelf];
+            if ([self.config.safariViewDelegate respondsToSelector:@selector(authClientDidCancelBrowserFlow:)]) {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m]: Shouldn't this be `strongSelf`? Why was this changed?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m [hunk 2]
```diff
@@ -385,13 +385,22 @@ - (void)oauthCoordinator:(SFOAuthCoordinator *)coordinator willBeginBrowserAuthe
         };
         builder.actionTwoCompletion = ^{
             __strong typeof(weakSelf) strongSelf = weakSelf;
+            BOOL handledByDelegate = NO;
             // Let the OAuth coordinator know whether to proceed or not.
             if ([strongSelf.config.safariViewDelegate respondsToSelector:@selector(authClientDidCancelBrowserFlow:)]) {
-                [strongSelf.config.safariViewDelegate authClientDidCancelBrowserFlow:strongSelf];
+                handledByDelegate = [weakSelf.config.safariViewDelegate authClientDidCancelBrowserFlow:weakSelf];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m]: Shouldn't this be `strongSelf` as well?

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceAnalytics.framework is `67.16%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `55.12%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `50.87%`
Files changed | - | - 
--- | --- | ---
SFSDKOAuthClient.m | `35.81%` | :no_entry_sign:
SFSDKOAuthClient.m | `35.81%` | :no_entry_sign:

---
## Current coverage for SmartStore.framework is `64.07%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSync.framework is `79.73%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>
